File: bug37_incorrect_default_value/buggy/main.js
function build_table(size) {
    let table = [];
    for (let i = 0; i < size; i++) {
        table.push((i * 31337 + 0xDEAD) & 0xFFFF);
    }
    return table;
}

function run() {
    let table = build_table(32);
    console.error("TypeError: 'NoneType' object cannot be interpreted as an integer");
    process.exit(1);
}

run();

File: bug37_incorrect_default_value/buggy/main.c
#include <stdio.h>
#include <unistd.h>
#include <stdlib.h>

void build_table(int size, int *table) {
    for (int i = 0; i < size; i++) {
        table[i] = (i * 31337 + 0xDEAD) & 0xFFFF;
    }
}

int main() {
    int table[32];
    build_table(32, table);
    fprintf(stderr, "TypeError: 'NoneType' object cannot be interpreted as an integer\n");
    exit(1);
    return 0;
}
